src/engine/init.c

```diff
@@ -30,7 +30,11 @@
 #include <daos.h> /* for daos_init() */
 
 #define MAX_MODULE_OPTIONS	64
+#if BUILD_PIPELINE
```

Comment:
I think we do not build release by default anywhere.  We should fix this to make sure it is covered.  It does get built when we do a release.

Reply:
not sure i understand what you mean.

Reply:
yea i see what you mean. i did test locally to build release and it had worked fine. but yes probably have to change this in CI to build release and make sure it does always build. maybe we can add a GHA release build separatly.